Sources/Segment/Utilities/Networking/HTTPSession.swift

```diff
@@ -0,0 +1,27 @@
+import Foundation
+
+#if os(Linux) || os(Windows)
+import FoundationNetworking
+#endif
+
+public protocol DataTask {
+  var state: URLSessionTask.State { get }
+  func resume()
+}
+
+// An enumeration of default `HTTPSession` configurations to be used
+// This can be extended buy consumer to easily refer back to their configured session.
+public enum DefaultHTTPSession {
```

Comment:
I don't know if I love this idea, but I wanted to capture something that could be used (and extended) to provide default configurations of different HTTPSessions. I would imagine in Arc we would extend this to provide something that hooks into the Chromium networking stack.